src/App.js

```diff
@@ -9,94 +9,97 @@ class App extends Component {
   constructor() {
     super();
     this.state = {
-      areGamesLoaded: false,
-      areGenresLoaded: false,
-      error: null,
+      errors: null,
       games: null,
       genres: null,
       popUpInfo: null,
-      popUpGenreID: null,
+      popUpGenre: null,
       searching: '',
       filteredGames: []
     }
   }
+  getData = (request) => {
+    const url = 'https://whateverly-datasets.herokuapp.com/api/v1/';
 
-  componentDidMount() {
-    fetch('https://whateverly-datasets.herokuapp.com/api/v1/games')
+    fetch(`${url + request}`)
       .then(data => data.json())
       .then(result => this.setState({
-        areGamesLoaded: true,
-        games: result.games
+        [request]: result[request]
       }))
-      .catch(error => {
+      .catch(errors => {
         this.setState({
-          error
+          errors
         })
       })
+  }
 
-
-    fetch('https://whateverly-datasets.herokuapp.com/api/v1/genres')
-      .then(data => data.json())
-      .then(result => this.setState({
-        areGenresLoaded: true,
-        genres: result.genres
-      }))
-      .catch(error => {
-        this.setState({
-          error
-        })
-      })
+  componentDidMount() {
+    this.getData('games');
+    this.getData('genres');
   }
 
-  createPopUp = (event, genreID) => {
+  createPopUp = (event) => {
+    const isCarousel = event.target.closest('.Carousel');
+    const popUpGenre = isCarousel && event.target.closest('.Carousel').dataset.genre;
+
     const popUpInfo = this.state.games
       .find(game => {
         return game.img === event.target.src;
-      })
-    this.setState({ popUpInfo, popUpGenreID: genreID });
+      });
+
+    this.setState({ popUpInfo, popUpGenre });
   }
 
   closePopUp = () => {
-    this.setState({ popUpInfo: null })
+    this.setState({
+      popUpInfo: null,
+      popUpGenre: null,
+    })
   }
 
   checkFilterInput = (event) => {
     let inputValue = event.target.value.toLowerCase();
-    this.setState({searching: inputValue});
-    let filteredGames = this.state.games
+    const filteredGames = this.state.games
       .filter(game => game.game.toLowerCase().includes(inputValue))
-      .sort((a,b) => {
-        if (a.game > b.game){return 1} 
-        if (a.game < b.game) {return -1} 
-        return 0;})
-    this.setState({filteredGames: filteredGames});
+      .sort((a, b) => a.game.localeCompare(b.game));
+
+    this.closePopUp();
+    this.setState({
+      searching: inputValue,
+      filteredGames
+    });
```

Comment:
Consolidated to calls to setState into one, changed our method of filtering to use an internal method instead of our custom one.